delivery_sendcloud_oca/models/sendcloud_shipping_method_country.py

```diff
@@ -0,0 +1,143 @@
+        string="Specific Delivery Product",
+        domain="[('type', '=', 'service')]",
+        help="This product will be used on the sale order line",
+        readonly=False,
+    )
+    company_id = fields.Many2one("res.company", required=True)
+    enable_price_custom = fields.Boolean(
+        compute="_compute_price_custom",
+        inverse="_inverse_price_custom",
+        readonly=False,
+    )
+    price_custom = fields.Float(
+        compute="_compute_price_custom",
+        inverse="_inverse_price_custom",
+        readonly=False,
+        string="Custom Price",
+        help="This price will override the standard price and will be applied "
+        "to the shipping price.",
+    )
+    price_check = fields.Selection(
+        [
+            ("standard", "Standard"),
+            ("custom", "Custom"),
+            ("unavailable", "Unavailable"),
+        ],
+        compute="_compute_price_custom",
+    )
+
+    @api.depends("iso_2", "company_id", "method_code")
+    def _compute_price_custom(self):
+        isos = self.mapped("iso_2")
+        companies = self.mapped("company_id")
+        method_codes = self.mapped("method_code")
+        custom_items = self.env["sendcloud.shipping.method.country.custom"].search(
+            [
+                ("iso_2", "in", isos),
+                ("company_id", "in", companies.ids),
+                ("method_code", "in", method_codes),
+            ],
+        )
+        for item in self:
+            custom = custom_items.filtered(
+                lambda ci, i=item: ci.iso_2 == i.iso_2
+                and ci.method_code == i.method_code
+                and ci.company_id == i.company_id
+            )
+            if custom:
+                item.price_check = "custom"
+                item.enable_price_custom = custom[0].enable_price_custom
+                if custom[0].enable_price_custom:
+                    item.price_custom = custom[0].price
+                else:
+                    item.price_custom = item.price
+                item.product_id = custom[0].product_id
+            else:
+                item.price_custom = item.price
+                item.price_check = "standard"
+
+    def _inverse_price_custom(self):
+        for item in self:
+            shipping_method_country = self.env[
+                "sendcloud.shipping.method.country.custom"
+            ].search(
+                [
+                    ("iso_2", "=", item.iso_2),
+                    ("company_id", "=", item.company_id.id),
+                    ("method_code", "=", item.method_code),
+                ],
+                limit=1,
+            )
+            if shipping_method_country:
+                shipping_method_country.price = item.price_custom
+                shipping_method_country.product_id = item.product_id
+                shipping_method_country.enable_price_custom = item.enable_price_custom
+            else:
+                self.env["sendcloud.shipping.method.country.custom"].create(
```

Comment:
Build a list of create values in the loop, then call `create()` once.

Reply:
Done !